### DISCUSSION TITLE
Default basic settings

### DISCUSSION THREAD

**Discussion by lordbanastor** [Courseplay/Courseplay_FS22#3082] [Q&A]
I tried to find a related question in the forum, but I couldn't, so here goes: is there a way to set my own default basic settings in CP? 

e.g.: Whenever I buy a new vehicle, it always comes with some basic setting I have to change: 

Open Hud with mouse - activated -> deactivated
Allow using the pathfinder in turns - deactivated -> activated
Return to starting point - deactivated -> activated
Raise tools - early -> late
Lower tools - late -> early
Field speed - 20.0 kph -> 25.0kph
Turn speed - 8.0 kph -> 10.0 kph
Reverse speed - 5.0 kph -> 10.0 kph
Silo speed - 15.0 kph -> 10.0 kph

I wonder if there is a way to set my own settings for these (and other) particular settings so whenever I buy a new vwhicle they are already set to my preferences. Sorry if this information exists already somewhere else, I just couldn't find, so point me in the right direction, if you will. Thank you all.

**Comment by tn4799** [comment]
I've already had this thought as well. 
It might be possible though editing the xml config file, but I dont know for sure.

But I'm also working an solution to configure the default values for vehicles in the global settings. But I cannot estimate when I will be done with that.

**Reply by bojanh66** [reply]
It seems i am not the only one who wish for different default settings for each vehicle...i use almost the same settings except a bit higher speeds

**Reply by tn4799** [reply]
Well, different defaults would not be possible to be configured in the general settings. but that should be possible in the courseplay.xml in the modSettings. 

If I'm wrong with that I'm sorry for spreading misinformation.

**Comment by Tensuko** [comment]
> Open Hud with mouse - activated -> deactivated

-> remove mouse curson on keybinds for this, then this is global, as del key is always active.

> Allow using the pathfinder in turns - deactivated -> activated

-> Can definitly talk about have this as a base default.


> Return to starting point - deactivated -> activated

-> definitly something that is personal, that would be something for @tn4799 to consider for his Idea.

> Raise tools - early -> late Lower tools - late -> early 

Something I rly don't recomment, I almost never change that setting and have perfectly worked fields.
If you are sure your tool need this and you know you got space on the headland, then you can change that setting, but that we will never make a default. (you can palce that into your personal vehicleconfig.xml for that specific tool: https://github.com/Courseplay/Courseplay_FS22/wiki/The-VehicleConfigurations.xml-file. )
Thinking of this, a default personal setting .xml might be a solution (set a vehicle, and hit a button that will make that default for all tools).
But I rly don't recomment that. (Personal experience)

> Field speed - 20.0 kph -> 25.0kph Turn speed - 8.0 kph -> 10.0 kph Reverse speed - 5.0 kph -> 10.0 kph Silo speed - 15.0 kph -> 10.0 kph

Field speed, well, whatever. Turn speed however can be tricky, it will definitly screw up some turns with some tools when they drive too fast, I wonder if that small diference have any high time impact compared to the overall fieldwork, just that you keep that in mind.

For reverse speed, same problem, depends on the tool, can work better or worse.

Silo Speed, uhm yeah could also reduce that default to 10, I don't rly care.

**Reply by lordbanastor** [reply]
Some useful tips there, but I think maybe I wasn't clear with my point. I don't mean change the default settings from start, to everyone. I just mean some way to change the default settings for me, like AD, where I can set certain settings as default from the moment I change them. e.g.: from the moment I find out a setting is better than other, I can choose to set that setting as default from that moment on, so the next vehicles I buy will have the new settings instead of the previous one, and so forth...

**Reply by Tensuko** [reply]
I know what you want to do, the point is that this is extra effort anyway.
So I just wanted to give you some thoughts about your setting decisions, maybe you change your mind on some.
But if tn4799 want to check that out and he finds a way, can build that in, not for everything, but some settings sure, why not.
Have to see what can be done ;)

**Reply by lordbanastor** [reply]
Appreciate that. And thank you for keeping this mod relevant. This mod is so high in my ranking that if I couldn't have this mod in my game, I simply wouldn't play at all.
